### PR TITLE
Merge unstable deneb june 6th

### PR DESCRIPTION
**Pull request by realbigsean**
## Issue Addressed

Which issue # does this PR address?

## Proposed Changes

Please list or describe the changes introduced by this PR.

## Additional Info

Please provide any additional information. For example, future considerations
or information useful for reviewers.

### CODE REVIEW

**Comment by jimmygchen** [comment]
@realbigsean 
I made a quick attempt to merge latest unstable into this branch before I realized you may have already done some of the work on the 14th July branch!

The branch builds and passes lint, but I'm not sure about the correctness of the merge since it appeared to be a tricky one. I've added comments on places where I added additional code due to the `BeaconProcessor` refactor. Feel free to use it only if it helps!

https://github.com/realbigsean/lighthouse/pull/37

File: beacon_node/beacon_chain/src/beacon_chain.rs [hunk 1]
```diff
@@ -2798,6 +2831,9 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
             notify_execution_layer,
         )?;
 
+        //TODO(sean) error handling?
```
Comment (pawanjay176) [inline beacon_node/beacon_chain/src/beacon_chain.rs:2834]: Probably something that makes more sense in the original PR. I say we keep it as is here and change it only if required for something deneb specific.
Also, the errors are being handled at the calling site as well, so not too much of an issue imo.

**Review by pawanjay176 — APPROVED** [review]
LGTM. I mainly looked at changes in deneb stuff resulting from the merge and paid some extra attention to changes resulting from:
1. Broadcast validation PR
2. Changes in beacon processor because of moving it to a separate module

I'm satisfied with the changes. Amazing work with the changes, the conflicts would have been nasty.
Feel free to merge (i think there's one conflict remaining still)

**Comment by pawanjay176** [comment]
@realbigsean merged `deneb-free-blobs` to get CI going on this

File: beacon_node/beacon_chain/src/beacon_chain.rs (hunk 1 above)
Comment (realbigsean) [reply beacon_node/beacon_chain/src/beacon_chain.rs:2834]: Yea I think the thing here was I was wondering if I should pass the error through `handle_block_error`, but that method just prints out a log. I think `publish_fn` handles the logging fine, there's only a couple error cases and they're logged or returned via API errors

**Pull request merged**